### PR TITLE
Porting `CircuitComponent`s to develop

### PR DESCRIPTION
**Pull request by SamFerracin**
**Context:**
Adds `CircuitComponent` and tests. A few more changes to `Wires` (added `__eq__` and replaced `>>` with `@`) to make it work

### CODE REVIEW

File: tests/test_lab_dev/test_circuit_components.py [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+
+class TestCircuitComponent:
+    r"""
+    Tests ``CircuitComponent`` objects.
+    """
+
+    @pytest.mark.parametrize("x", [0.1, [0.2, 0.3]])
+    @pytest.mark.parametrize("y", [0.4, [0.5, 0.6]])
+    def test_init(self, x, y):
+        name = "my_component"
```
Comment (SamFerracin) [inline tests/test_lab_dev/test_circuit_components.py:41]: This (and all other cases where I initialize a Dgate from the triple) will all be replaced by, e.g., `Dgate(x, y)` when I add gates and states

File: mrmustard/lab_dev/wires.py [hunk 2]
```diff
@@ -331,12 +342,17 @@ def _outin(self_in: int, self_out: int, other_in: int, other_out: int) -> np.nda
...
-        A new Wires object with the wires of ``self`` and ``other`` combined as two
-        components in a circuit: the output of self connects to the input of other wherever
-        they match. All surviving wires are arranged in the standard order.
-        A ValueError is raised if there are any surviving wires that overlap.
+        A new ``Wires`` object with the wires of ``self`` and ``other`` combined.
+
+        The output of ``self`` connects to the input of ``other`` wherever they match. All
+        surviving wires are arranged in the standard order.
+
+        This function does not add missing adjoints.
```
Comment (sylviemonet) [inline mrmustard/lab_dev/wires.py:352]:
```suggestion
        This function only performs the contractions given and it does not add missing adjoints.
```

A small suggestion to be more clear on this.

**Review by sylviemonet — APPROVED** [review]
Nice PR. I think it looks good to me.

File: mrmustard/lab_dev/circuit_components.py [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+        r"""
+        The ``Wires`` in this component.
+        """
+        return self._component.wires.dual
+
+    def __repr__(self) -> str:
+        return repr(self._component)
+
+
+def add_bra(components: Sequence[CircuitComponent]) -> Sequence[CircuitComponent]:
```
Comment (ziofil) [inline mrmustard/lab_dev/circuit_components.py:322]: do we still need this? (also in __all__)

**Review by ziofil — APPROVED** [review]
looks good, just one function to remove probably

**Pull request merged**